skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,342 @@
+"""Adapters to ngboost regressors with probabilistic components."""
+# copyright: skpro developers, BSD-3-Clause License (see LICENSE file)
+
+__author__ = ["ShreeshaM07"]
+
+import numpy as np
+from sklearn.utils import check_random_state
+
+from skpro.regression.base import BaseProbaRegressor
+
+
+class NGBoostRegressor(BaseProbaRegressor):
+    """
+    Constructor for all NGBoost models.
+
+    This class implements the methods that are common to all NGBoost models.
+    Unless you are implementing a new kind of regression (e.g. interval-censored, etc.),
+    you should probably use one of NGBRegressor, NGBClassifier, or NGBSurvival.
+
+    Parameters
+    ----------
+        Dist              : assumed distributional form of Y|X=x.
```

Comment:
this is not in numpydoc format and renders badly. Kindly follow examples in other regressors.